src/Lexer.js

```diff
@@ -229,8 +250,21 @@ module.exports = class Lexer {
         continue;
       }
 
+      if (tokensLength = this.runTokenzierExtension(src, tokens, 'paragraph')) {
+        src = src.substring(tokensLength);
+        continue;
+      }
```

Comment:
do we need this check before each tokenizer?

Reply:
Yes we do. I just didn't apply it everywhere yet in case we want to keep changing things. Should I go ahead and copy it around?

Reply:
Is it worth it to cache whether `this.options.extensions` exists in the constructor maybe? Is it faster to check `if(haveExtensions)` than `if(this.options.extensions)` ?

Reply:
I copied this code everywhere now. Running benchmarks...

Reply:
Moving the check does help with the slowdown some. ~ `4814ms` to `4718ms` on es6 commonmark spec.